### DISCUSSION TITLE
Z-WAVE-SERVER: The node failed to decode the message. (ZW1405)

### DISCUSSION THREAD

**Discussion by kristof-mattei** [zwave-js/zwave-js#7410] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?

After sending 10+ dimming signals to a ZEN77, the device ceases to respond, and my whole Z-Wave system goes down for a while. 

Then, as updates are slowly reacted to the logs are full of `ZWaveError: The node failed to decode the message. (ZW1405)`.

The 'other' logs inside of the app (attached) also show `[Node 025] failed to decode the message, retrying with SPAN extension...` 

What did you expect to happen instead?

That the system stays up

  1. Get a ZEN77
  2. Via Home-Assistant -> HomeKit 'swipe' the dimmer from 0 -> 100 -> 0 -> 100.
  3. See ZWave falling over.

Anything else we should know?

1. Issues occur on `Node 025`.
2. I have another ZEN77 which does _NOT_ have this issue... 

### Software versions

Driver (node-zwave-js): 14.3.2

Z-Wave JS UI: 9.27.2.b0ade31 (Also experienced on 9.26.0)

Home Assistant Z-Wave Integration: 2024.11.2

Home Assistant Z-Wave JS Addon: 0.58.1

If you are using something non-standard, tell us here: ...

### Z-Wave Controller (Stick/Dongle/...)

Aeotec Z-Stick 7 FW v7.21.5, but issue was before this FW too.

### Device information

Manufacturer: ZOOZ
Model name: ZEN77
Node ID: 025

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_current.log](https://github.com/user-attachments/files/17782698/zwavejs_current.log)

**Comment by AlCalzone** [comment]
Interesting case... It appears that the issues you describe are caused by a temporary connectivity problem, during which both the controller and the node retry their communication and go out of sync. The continuous onslaught of commands causes a bit of trouble during that process.
I've started working on ways to make the sequence of commands sent by Z-Wave JS more predictable, but it's not quite there yet, so this will take a while to be solved.

Aside from that:
If you haven't already, I recommend taking a look at https://zwave-js.github.io/node-zwave-js/#/troubleshooting/first-steps, except for the part "Reduce reports from the network", which does not seem to be the issue here.

Is this swiping a typical usage pattern or did you just do it this way to demonstrate the problem? Because sending lots of commands like this can be problematic in Z-Wave, and there are better ways to start and stop dimming that just take 2 commands total.

**Reply by kristof-mattei** [reply]
When exposing a dimmer in Home-Assistant there is a slider:

<img src="https://github.com/user-attachments/assets/00f86b27-ac9a-4530-b1c2-1fa6ccbe7f0a" width="300">


Moving the slider up/down doesn't send signals. It merely sends a signal with the 'selected' value when you release the slider.

This is different from iOS. When you move the slider the light changes brightness real-time. 

<img src="https://github.com/user-attachments/assets/3b776d9b-6082-4958-8e92-d2e7c84c7f99" width="300">


(I just realized that my images actually don't show anything useful).


Now, here's the thing, I have another ZEN77, MORE lightbulbs, same FW, which does not exhibit this behavior. In iOS I can go from 0->100->0->100 and again and again and again and see the lights dim and brighten real-time. 

That one sits closer to the controller though.

So I'm thinking:

* HW error?
* Bulb error?
* Distance to stick?
* Maybe we're sending messages too fast?

Would it help if I move the Z-Wave stick next to the 'faulty' dimmer and see if I can reproduce like that?

I'm also gonna review that link you sent and see what info I can get from that.

**Reply by AlCalzone** [reply]
Essentially, it's a combination of:
- a less-than-perfect connection between the Z-Wave stick and the device
- sending commands too quickly for handling potential errors
- Z-Wave JS having trouble with handling those errors

The first point may be something you can fix, the others are things I have to improve on our side.
